Rework secret-handshake exercise docs (#6)
kytrinyx/problem-specifications: exercises/secret-handshake/introduction.md

# Introduction

There's a joke that goes:

_There are 10 types of people in the world._
_There are those who understand binary, and those who don't._

You belong to the group that understands binary, and you and your binary-savvy friends want to create a secret handshake.
You don't want anyone who isn't in the know to be able to crack the code.

> So you come up with a sequence of actions, where each action corresponds to a different binary digit.

Reviewer: Can we add a exercism/note that says "No idea what binary is or how to use it? Check this tutorial..."

Author: I searched for something by Vaidehi, and found this: https://medium.com/basecs/bits-bytes-building-with-binary-13cb4289aafa

Look good?

> There's a joke that goes:
> 
> _There are 10 types of people in the world._
> _There are those who understand binary, and those who don't._
> 
> You belong to the group that understands binary, and you and your binary-savvy friends want to create a secret handshake.
> You don't want anyone who isn't in the know to be able to crack the code.
> 
> So you come up with a sequence of actions, where each action corresponds to a different binary digit.

Reviewer: I feel like binary is an implementation detail, not the story. And so I feel like the joke is maybe unnecessary and adds noise. So I suggest getting rid of that and focussing a little more on the fun story angle.

Suggestion:

```suggestion
You are making a secret coding club with some friends and friends-of-friends. 
You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
You don't want anyone who isn't in the know to be able to crack the code.

You and your friends have decided to come up with a code that use binary (because you're a geek and think binary is cool) and turns it into a series of actions that you do during the handshake.
```

Author: Nice, yeah. It's so funny how I even failed to consider that we could dump the binary thing.

# Introduction

> You are making a secret coding club with some friends and friends-of-friends.

Reviewer: ```suggestion
You are starting a secret coding club with some friends and friends-of-friends.
```

You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
You don't want anyone who isn't in the know to be able to crack the code.

> You designed the code so that it takes a binary number and turns it into a series of actions.

Reviewer: ```suggestion
You designed the code so that one person says a number between 1 and 30, and the other other person turns it into a series of actions.
```

Author: ```
> 31.to_s(2)
=> "11111"
> 30.to_s(2)
=> "11110"
```

Author: Fixed this.

> The secret handshake is created by taking a binary number and turning it into a series of actions.

Reviewer 2: "is created by taking a binary number"

Maybe rephrase slightly to make it seem more, eh, designed? As in: you and friends have designed a binary coding scheme or something like that.

```exercism/note

> If you aren't sure how binary works, check out [this tutorial][intro-to-binary].

Reviewer: ```suggestion
If you aren't sure what binary is or how it works, check out [this tutorial][intro-to-binary].
```

Reviewer: Suggest moving this to instructions, and removing all reference to binary out of introduction.

# Introduction

You are starting a secret coding club with some friends and friends-of-friends.

> You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.

Reviewer: Just one nit to stop every sentence starting with "you"

```suggestion
Not everyone knows each other, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
```

You don't want anyone who isn't in the know to be able to crack the code.

> You've designed the code so that one person says a number between 1 and 31, and the other other person turns it into a series of actions.

Reviewer 2: I like this new version! We no longer mention anything about using a bit-wise encoding scheme though. That was intentional?

Author: Yeah, we moved all the bits (no pun intended) about binary into the instructions, leaving the introduction only as the scenario that puts the exercise into context.

> You designed the code so that one person says a number between 1 and 31, and the other other person turns it into a series of actions.

Reviewer: ```suggestion
You've designed the code so that one person says a number between 1 and 31, and the other other person turns it into a series of actions.
```
